app/controllers/authorships_controller.rb

```diff
@@ -221,7 +221,8 @@ def get_publication_via_sciencewire!(sw_id)
   # @return [Publication]
   def get_publication_via_wos!(author, wos_uid)
     WebOfScience.harvester.author_uid(author, wos_uid)
-    pub = Publication.find_by(wos_uid: wos_uid)
+    # we find the publication by looking for any matching publication for this WOS record (could be MEDLINE or WOS ID)
+    pub = WebOfScienceSourceRecord.find_by(uid: wos_uid)&.record&.matching_publication
```

Comment:
This set of conditionals is hard to understand.  Can you make the logic do something like this, so we better understand where the nils might come from?
```ruby
wossr = WebOfScienceSourceRecord.find_by(uid: wos_uid)
if wossr
  pub = wossr.record.matching_publication
...
```

Reply:
Yeah, ok.  I don't think we needed to check for the second nil (a record should return a matching publication), but I'll break things out with different error messages based on what happened.